kernel/arch/dreamcast/include/dc/g2bus.h

```diff
@@ -31,50 +32,184 @@
-#define BBA_DMA2_MODE   3
-#define BBA_DMA2_G2CHN  2
-#define BBA_DMA2_SHCHN  0
+    Functions that act as callbacks when G2-DMA completes should be of this type.
+    These functions will be called inside an interrupt context, so don't try to
+    use anything that might stall.
 
+    \param  data            User data passed in to the g2_dma_transfer()
+                            function.
+*/
 typedef void (*g2_dma_callback_t)(ptr_t data);
-int g2_dma_transfer(void *from, void * dest, uint32 length, int block,
+
+/** \brief  Perform a DMA transfer between SH-4 RAM and G2 Bus
+
+    This function copies a block of data between SH-4 RAM and G2 Bus via DMA. 
+    You specify the direction of the copy (SH4TOG2BUS/G2BUSTOSH4). There are all 
+    kinds of constraints that must be fulfilled to actually do this, so
+    make sure to read all the fine print with the parameter list.
+
+    If a callback is specified, it will be called in an interrupt context, so
+    keep that in mind in writing the callback.
+
+    \param  sh4             Where to copy from/to. Must be 32-byte aligned.
+    \param  g2bus           Where to copy from/to. Must be 32-byte aligned.
+    \param  length          The number of bytes to copy. Must be a multiple of
+                            32.
+    \param  block           Non-zero if you want the function to block until the
+                            DMA completes.
+    \param  callback        A function to call upon completion of the DMA.
+    \param  cbdata          Data to pass to the callback function.
+    \param  dir             SH4TOG2BUS or G2BUSTOSH4.
+    \param  mode            Any value; Here for compatibility.
+    \param  g2chn           See g2b_channels.
+    \param  sh4chn          Any value; Here for compatibility.
+    \retval 0               On success.
+    \retval -1              On failure. Sets errno as appropriate.
+
+    \par    Error Conditions:
+    \em     EINPROGRESS - DMA already in progress \n
+    \em     EFAULT - sh4 and/or g2bus is not 32-byte aligned \n
+    \em     EINVAL - Invalid g2chn
+    \em     EIO - I/O error
+
+    \see    g2b_channels
+*/
+int g2_dma_transfer(void * sh4, void * g2bus, uint32_t length, uint32_t block,
                     g2_dma_callback_t callback, ptr_t cbdata,
-                    uint32 dir, uint32 mode, uint32 g2chn, uint32 sh4chn);
+                    uint32_t dir, uint32_t mode, uint32_t g2chn, uint32_t sh4chn);
+
+/** \brief  Initialize DMA support.
+
+    This function sets up the DMA support for transfers to/from the G2 Bus.
+
+    \retval 0               On success (no error conditions defined).
+*/
+int g2_dma_init(void);
+
+/** \brief  Shutdown DMA support. */
+void g2_dma_shutdown(void);
+
+/** \brief  G2 context
+
+    A G2 context containing the states of IRQs and G2 DMA. This struct
+    is used in g2_(un)lock()'s
+*/
+typedef struct { 
+    uint32_t irq_state;
+} g2_ctx_t;
+
+/* Internal constants */
```

Comment:
```suggestion
/* Internal constants */
/** \cond */ 
```
This is how you flag something that's internal to not get indexed by Doxygen if it's not part of the public API... you use `\cond` to start a block and `\endcond` to end it.